### ISSUE TITLE
a fresh install of vcpkg and visual studio 2019 does not find header files

### ISSUE THREAD

**Issue by terrylyons** [microsoft/vcpkg#9432]
**Describe the bug**
I have installed the dependencies to a simple C++ project using vcpkg with x86 and x64 triplets. Previously, I installed vcpkg and the linkage to visual studio exactly as in the docs. However build of the sample code fails (see below) claiming not to find the header files. 

I note that the environmental variable in visual studio VCLibPackagePath is set, but points to an empty folder vcpkgs\vcpkg.user below APPDATA  while the real data seems to be held below the following path: C:\Users\xxxxxx\source\vcpkg\packages

Th installation is a fresh one with latest versions. 

**Environment**
- OS: Windows 10.0.18362, VS 2019, clean new environment. 
- Compiler: revision
Microsoft Visual Studio Enterprise 2019
Version 16.4.2
VisualStudio.16.Release/16.4.2+29613.14

**To Reproduce**
1. 
vcpkg install unittest-cpp boost-thread boost-iostreams mpir
vcpkg.exe install boost-thread:x64-windows boost-iostreams:x64-windows unittest-cpp:x64-windows mpir:x64-windows

Builds and installs the appropriate libraries etc. 

2. See error
However on building the project that expects these dependencies visual studio produces errors like
>LatticePathTests.cpp
1>C:\Users\tlyons\source\repos\terrylyons\libalgebra_tests\libalgebra\libalgebra\libalgebra.h(53,10): fatal error C1083: Cannot open include file: 'boost/thread/shared_mutex.hpp': No such file or directory
1>LibAlgebraUnitTests.cpp
1>C:\Users\tlyons\source\repos\terrylyons\libalgebra_tests\LibAlgebraUnitTests\LibAlgebraUnitTests.cpp(5,10): fatal error C1083: Cannot open include file: 'UnitTest++/UnitTest++.h': No such file or directory
1>HallSetTests.cpp
1>C:\Users\tlyons\source\repos\terrylyons\libalgebra_tests\libalgebra\libalgebra\libalgebra.h(53,10): fatal error C1083: Cannot open include file: 'boost/thread/shared_mutex.hpp': No such file or directory
1>makebm.cpp
1>memfile.cpp
1>C:\Users\tlyons\source\repos\terrylyons\libalgebra_tests\LibAlgebraUnitTests\memfile.h(3,10): fatal error C1083: Cannot open include file: 'boost/iostreams/device/mapped_file.hpp': No such file or directory

**Expected behavior**
A successful build was expected.

**Failure logs** 
-(please attached failure logs)

**Additional context**

**Comment by NancyLi1013** [comment]
Hi @terrylyons thanks for posting this issue.
It seems that you can install these libs successfully and this issue appeared when you used them. 
If so, could you help provide the project case and detail steps to reproduce?
Besides this, have you integrated with VS by the command` ./vcpkg integrate install`?

**Comment by terrylyons** [comment]
Hi @NancyLi1013, I worked from the command window using runas:domain\administrator cmd and not from powershell. So I used syntax like "vcpkg integrate install". 

However, on return to this problem after your request, I ran the exact command you suggested from powershell in my user account (unelevated), and bingo!! it all worked as it should (dlls magically loaded etc). So perhaps integration is affected by the choices of elevated and\or which account does what... 

Hope this helps.

**Comment by NancyLi1013** [comment]
We suggest use powershell to install ports and integrate with VS.
Please refer to [readme](https://github.com/microsoft/vcpkg#tab-completion--auto-completion) and [doc](https://github.com/microsoft/vcpkg/blob/master/docs/users/integration.md#integrate-command).

**Comment by shelper** [comment]
I had exactly the same issue, and i run both in poweshell and cmd window with `vcpkg integrate install` but still could not find the packages for visual studio....

In the `cl.command.1.tlog` file in my project folder, i found the line like below:
`/I<path_to_vcpkg>\VCPKGINCLUDE`
but what should be there is 
`/I<path_to_vcpkg>\VCPKG\installed\os_type\INCLUDE`

i am using visual studio 2019 version `16.5.2` and vcpkg the latest from github.

Tried everything including reinstall `vcpkg`. or reinstall `visual studio` or different packages, but none worked....

**Comment by NancyLi1013** [comment]
Hi @shelper 
Thanks for posting this issue.
Could you please help confirm if you encountered this issue when you used it?
Please also post the failures about this.

**Comment by shelper** [comment]
my bad, i think the integration works with visual studio, but not visual studio code.... sorry about that

**Comment by NancyLi1013** [comment]
Yes, Visual Studio Code is currently not supported yet.

**Comment by rhalbersma** [comment]
Please add vcpkg integration with VS Code!!

**Comment by NancyLi1013** [comment]
@rhalbersma 
As for the support for VS code, let's track this here #10217.

**Comment by Chris45215** [comment]
I have encountered the same problem, but only with VS2019 and not VS2017, with VCPKG version 2020.11.03, Visual Studio Community 2019 version 16.7.4, and Visual Studio Community 2017 version 15.9.14. I have used the Integrate command in both Powershell and command line, and in both administrator mode and non-administrator mode for both. 

I tested it all again when following the video tutorial at https://www.youtube.com/watch?v=y41WFKbQFTw, because that exact video is linked in the official documentation at https://github.com/Microsoft/vcpkg#quick-start-windows (the last line in the "Examples" section).
Specifically, in VS2019 when I type "#include <SDL2/SDL.h>", I get the error "cannot open source file "SDL2/SDL.h"."
But when I try the exact same line in my VS2017 install, it autocompletes and compiles correctly. 

I would greatly prefer to use VS2019 for my project, as some other libraries I need have very little compatibility with VS2017

**Comment by NancyLi1013** [comment]
@Chris45215 

I can find `SDL2/SDL.h` with Visual Studio 2019 16.9.1 in cmake project.

Please open a new issue if you still have this problem.

**Comment by Chris45215** [comment]
Thanks @NancyLi1013, I was able to get it too work too - but only after I re-downloaded the entirety of PCL with vcpkg and used the instruction ".\vcpkg install pcl:x64-windows", instead of ".\vcpkg install pcl:x64-windows-static" which I had used earlier. I can make the new issue, but I'm not sure if it was just weirdness on my side, or if there is a significant difference between the static and non-static version (I can't find any description of the difference). I've only recently learned about very significant differences between the vcpkg version and the "compile from source" version (https://pcl-tutorials.readthedocs.io/en/master/compiling_pcl_windows.html), as I need to use the visualization libraries but those seem to be excluded from the vcpkg version. 

I tried to use the AllInOne install initially, but it aborted because PATH variables were too long. So..... I've spent several days trying to get everything working, and this is quite a hill to climb. I really appreciate the help you provide, but I'm now on my 3rd completely different install method, and the downloads page for the dependencies in the "compile from source" guide doesn't even load (so I'm using https://gist.github.com/UnaNancyOwen/59319050d53c137ca8f3#file-pcl1-8-1-md to tell me the versions of the dependencies I need).... I'm at the point of questioning and doubting everything I'm doing. I've not made new issues because there is clearly a lot I don't know, and I don't want to send other people on a wild goose chase if I poorly document a problem.

Is there some easier way to install this that I somehow overlooked? I can't believe it's supposed to be this difficult just to get started.

**Comment by NancyLi1013** [comment]
@Chris45215 

Thanks for your above details.
For static and dynamic, there are some differences between these two builds in usage.
Especially, for static build, there is some extra work that needs to be done.
Please see the doc here https://devblogs.microsoft.com/cppblog/vcpkg-updates-static-linking-is-now-available/
